pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java

```diff
@@ -714,10 +712,7 @@ private CompletableFuture<Void> provisionPartitionedTopicPath(int numPartitions,
     }
 
     protected CompletableFuture<SchemaCompatibilityStrategy> getSchemaCompatibilityStrategyAsync() {
-        return validateTopicPolicyOperationAsync(topicName,
-                PolicyName.SCHEMA_COMPATIBILITY_STRATEGY,
-                PolicyOperation.READ)
-                .thenCompose((__) -> getSchemaCompatibilityStrategyAsyncWithoutAuth()).whenComplete((__, ex) -> {
+        return getSchemaCompatibilityStrategyAsyncWithoutAuth().whenComplete((__, ex) -> {
```

Comment:
`getSchemaCompatibilityStrategyAsync` and `getSchemaCompatibilityStrategyAsyncWithoutAuth` contain about the same logic now.
Please get rid of `getSchemaCompatibilityStrategyAsyncWithoutAuth`.
It seems that replacing `getSchemaCompatibilityStrategyAsyncWithoutAuth` with `getSchemaCompatibilityStrategyAsync` in `SchemasResourcesBase.postSchemaAsync`  and then inlining `getSchemaCompatibilityStrategyAsyncWithoutAuth` into `getSchemaCompatibilityStrategyAsync` would address the problem.